job.py

```diff
@@ -0,0 +1,85 @@
+import shutil
+from urllib.request import urlopen
+from pathlib import Path
+import ssl
+import json
+from logger import logger
+
+
+def coroutine(f):
+    def wrap(*args, **kwargs):
+        gen = f(*args, **kwargs)
+        gen.send(None)
+        return gen
+    return wrap
+
+
+def get_and_write_data(condition, url):
+    context = ssl._create_unverified_context()
+    file = 'punchline.txt'
+    with condition:
+        with urlopen(url, context=context) as req:
+            resp = req.read().decode("utf-8")
+            resp = json.loads(resp)
+        if req.status != 200:
+            raise Exception(
```

Comment:
Лучше рейзить конкретный тип ошибки, например 400ю, можно наследоваться и сделать свой класс, например ResourceNotFoundError:
```suggestion
            raise ResourceNotFoundError("Cannot fetch data")
```